### PR TITLE
fix: Skip fileResource.issAssigned check for CREATE

### PR DESCRIPTION
**Pull request by abyot**

### CODE REVIEW

**Review by stian-sandvold — APPROVED** [review]

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/EventDataValuesValidationHook.java [hunk 1]
```diff
@@ -189,7 +189,11 @@ private void validateFileNotAlreadyAssigned( ValidationErrorReporter reporter, E
         FileResource fileResource = reporter.getValidationContext().getFileResource( dataValue.getValue() );
 
         reporter.addErrorIfNull( fileResource, event, E1084, dataValue.getValue() );
-        reporter.addErrorIf( () -> fileResource != null && fileResource.isAssigned(), event,
-            E1009, dataValue.getValue() );
+
+        if ( reporter.getValidationContext().getProgramStageInstance( event.getEvent() ) == null )
```
Comment (enricocolasante) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/EventDataValuesValidationHook.java:193]:
```suggestion
        if ( reporter.getBundle().getStrategy(event).isCreate() ) {
```

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/TrackedEntityAttributeValidationHook.java [hunk 2]
```diff
@@ -216,6 +216,10 @@ protected void validateFileNotAlreadyAssigned( ValidationErrorReporter reporter,
         FileResource fileResource = reporter.getValidationContext().getFileResource( attr.getValue() );
 
         reporter.addErrorIfNull( fileResource, te, E1084, attr.getValue() );
-        reporter.addErrorIf( () -> fileResource != null && fileResource.isAssigned(), te, E1009, attr.getValue() );
+
+        if ( reporter.getValidationContext().getTrackedEntityInstance( te.getTrackedEntity() ) == null )
```
Comment (enricocolasante) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/TrackedEntityAttributeValidationHook.java:220]:
```suggestion
        if ( reporter.getBundle().getStrategy( te ).isCreate() ) {
```

**Review by enricocolasante — CHANGES_REQUESTED** [review]
You can consider using the strategy to check if it is a create, it makes the code more explicit.

Anyway, you need to merge master into your branch as the ValidationContext was removed in a previous PR. I am really not sure why github is not showing this as a problem

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/EventDataValuesValidationHook.java (hunk 1 above)
Comment (abyot) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/EventDataValuesValidationHook.java]: fixed.

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/TrackedEntityAttributeValidationHook.java (hunk 2 above)
Comment (abyot) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/TrackedEntityAttributeValidationHook.java]: fixed.

**Review by enricocolasante — APPROVED** [review]

**Pull request merged**